File: Suhyun/Level1/신고 결과 받기/Solution.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public static void main(String[] args) {
        // 하나의 출력 인덱스
        String[] id_list = { "muzi", "frodo", "apeach", "neo" };
        String[] report = { "muzi frodo", "apeach frodo", "frodo neo", "muzi neo", "apeach muzi" };
        int k = 2;

        Map<String, Integer> id = new HashMap<String, Integer>();

        // 중복제거용
        String[] stream = Arrays.stream(report).distinct().toArray(String[]::new);
        // System.out.println(Arrays.toString(stream));

        for (int i = 0; i < id_list.length; i++) {

            id.put(id_list[i], 0);
        }
        Map<String, Integer> id_mail = new HashMap<String, Integer>();
        id_mail.putAll(id);

        for (int i = 0; i < stream.length; i++) {
            String[] wakare = stream[i].split(" ");
            id.put(wakare[1], id.get(wakare[1]) + 1);
            // if (id.get(wakare[1]) >= k) {
            // // id_mail.put(wakare[1], id_mail.get(wakare[1]) + 1);
            // id_mail.put(wakare[1], id_mail.get(wakare[1]) + 1);
            // }
        }

        for (int i = 0; i < stream.length; i++) {
            String[] wakare = stream[i].split(" ");

            if (id.get(wakare[1]) >= k) {
                // id_mail.put(wakare[1], id_mail.get(wakare[1]) + 1);
                id_mail.put(wakare[0], id_mail.get(wakare[0]) + 1);
            }

        }
        int[] answer = new int[id_list.length];
        for (int i = 0; i < id.size(); i++) {
            answer[i] = id_mail.get(id_list[i]);
        }

        System.out.println(Arrays.toString(answer));
    }
}

File: Suhyun/Level1/신고 결과 받기/Solution.js
let id_list = ["muzi", "frodo", "apeach", "neo"];
let report = ["muzi frodo", "muzi frodo", "apeach frodo", "frodo neo", "muzi neo", "apeach muzi"];
let k = 2;

let answer = [];
let id = new Map();
id_list.map(e => id.set(e, 0));
let mail_id = new Map(id);


//중복제거 
const unique = {};
report.forEach(el => {
    unique[el] = true;
})
const arr_unique = Object.keys(unique);


for (let i = 0; i < arr_unique.length; i++) {
    let wakare = arr_unique[i].split(" ");

    id.set(wakare[1], id.get(wakare[1]) + 1);

}
for (let i = 0; i < arr_unique.length; i++) {
    let wakare = arr_unique[i].split(" ");
    if (id.get(wakare[1]) >= k) {
        mail_id.set(wakare[0], mail_id.get(wakare[0]) + 1);

    }
}

for (let i = 0; i < id_list.length; i++) {
    answer.push(mail_id.get(id_list[i]));
}





return answer;
